snapshot/lib/snapshot/test_command_generator.rb

```diff
@@ -78,9 +78,17 @@ def find_device(device_name, os_version = Snapshot.config[:ios_version])
         end
       end
 
-      def device_udid(device_name, os_version = Snapshot.config[:ios_version])
+      def device_udid(device_name)
+        os = device_name =~ /^Apple TV/ ? "tvOS" : "iOS"
+        os_version = Snapshot.config[:ios_version] || Snapshot::LatestOsVersion.version(os)
+
         device = find_device(device_name, os_version)
 
+        if device.nil?
+          UI.user_error!("No device found named '#{device_name}' for version '#{os_version}'")
```

Comment:
Do I see correctly that it is re-trying to run it, even though there is no way to actually find the device on the second try, right? 
Ideally it should immediately fail the current device/language combination without retrying